### PR TITLE
Update to upstream v1.58.0

### PR DESCRIPTION
**Pull request by jen20**
This PR updates to version 1.58.0 of upstream.

This is the first version using `gomod-override` for generating `Gopkg.toml`, and brings in the latest `pulumi-terraform`, improving all the examples - hence the high number of files modified.

### CODE REVIEW

File: sdk/go/aws/ec2/peeringConnectionOptions.go [hunk 1]
```diff
@@ -20,120 +20,8 @@ import (
...
-//   requester {
-//     allow_vpc_to_remote_classic_link = true
-//     allow_classic_link_to_remote_vpc = true
-//   }
-// }
-// ```
 // 
 // Basic cross-account usage:
-// 
-// ```hcl
```
Comment (jen20) [inline sdk/go/aws/ec2/peeringConnectionOptions.go]: It's unclear to me why we're losing these examples with v0.4.6 of tf2pulumi. @pgavlin any ideas?

Comment (pgavlin) [reply sdk/go/aws/ec2/peeringConnectionOptions.go]: I don't think that this is a `tf2pulumi` thing: I think it's a `pulumi-terraform` thing. We'll probably pick these up as converted examples once the latest version of `pulumi-terraform` is rolled out.

Comment (pgavlin) [reply sdk/go/aws/ec2/peeringConnectionOptions.go]: Indeed, here's the diff I get for the Node version of that file with a newer `pulumi-terraform`:

```diff
diff --git a/sdk/nodejs/ec2/peeringConnectionOptions.ts b/sdk/nodejs/ec2/peeringConnectionOptions.ts
index 3247ca2..b843ae5 100644
--- a/sdk/nodejs/ec2/peeringConnectionOptions.ts
+++ b/sdk/nodejs/ec2/peeringConnectionOptions.ts
@@ -17,120 +17,34 @@ import * as utilities from "../utilities";
  * 
  * Basic usage:
  * 
- * ```hcl
- * resource "aws_vpc" "foo" {
- *   cidr_block = "10.0.0.0/16"
- * }
- * 
- * resource "aws_vpc" "bar" {
- *   cidr_block = "10.1.0.0/16"
- * }
- * 
- * resource "aws_vpc_peering_connection" "foo" {
- *   vpc_id      = "${aws_vpc.foo.id}"
- *   peer_vpc_id = "${aws_vpc.bar.id}"
- *   auto_accept = true
- * }
- * 
- * resource "aws_vpc_peering_connection_options" "foo" {
- *   vpc_peering_connection_id = "${aws_vpc_peering_connection.foo.id}"
- * 
- *   accepter {
- *     allow_remote_vpc_dns_resolution = true
- *   }
- * 
- *   requester {
- *     allow_vpc_to_remote_classic_link = true
- *     allow_classic_link_to_remote_vpc = true
- *   }
- * }
+ * ```typescript
+ * import * as pulumi from "@pulumi/pulumi";
+ * import * as aws from "@pulumi/aws";
+ * 
+ * const bar = new aws.ec2.Vpc("bar", {
+ *     cidrBlock: "10.1.0.0/16",
+ * });
+ * const fooVpc = new aws.ec2.Vpc("foo", {
+ *     cidrBlock: "10.0.0.0/16",
+ * });
+ * const fooVpcPeeringConnection = new aws.ec2.VpcPeeringConnection("foo", {
+ *     autoAccept: true,
+ *     peerVpcId: bar.id,
+ *     vpcId: fooVpc.id,
+ * });
+ * const fooPeeringConnectionOptions = new aws.ec2.PeeringConnectionOptions("foo", {
+ *     accepter: {
+ *         allowRemoteVpcDnsResolution: true,
+ *     },
+ *     requester: {
+ *         allowClassicLinkToRemoteVpc: true,
+ *         allowVpcToRemoteClassicLink: true,
+ *     },
+ *     vpcPeeringConnectionId: fooVpcPeeringConnection.id,
+ * });
  * ```
  * 
  * Basic cross-account usage:
- * 
- * ```hcl
- * provider "aws" {
- *   alias = "requester"
- * 
- *   # Requester's credentials.
- * }
- * 
- * provider "aws" {
- *   alias = "accepter"
- * 
- *   # Accepter's credentials.
- * }
- * 
- * resource "aws_vpc" "main" {
- *   provider = "aws.requester"
- * 
- *   cidr_block = "10.0.0.0/16"
- * 
- *   enable_dns_support   = true
- *   enable_dns_hostnames = true
- * }
- * 
- * resource "aws_vpc" "peer" {
- *   provider = "aws.accepter"
- * 
- *   cidr_block = "10.1.0.0/16"
- * 
- *   enable_dns_support   = true
- *   enable_dns_hostnames = true
- * }
- * 
- * data "aws_caller_identity" "peer" {
- *   provider = "aws.accepter"
- * }
- * 
- * # Requester's side of the connection.
- * resource "aws_vpc_peering_connection" "peer" {
- *   provider = "aws.requester"
- * 
- *   vpc_id        = "${aws_vpc.main.id}"
- *   peer_vpc_id   = "${aws_vpc.peer.id}"
- *   peer_owner_id = "${data.aws_caller_identity.peer.account_id}"
- *   auto_accept   = false
- * 
- *   tags = {
- *     Side = "Requester"
- *   }
- * }
- * 
- * # Accepter's side of the connection.
- * resource "aws_vpc_peering_connection_accepter" "peer" {
- *   provider = "aws.accepter"
- * 
- *   vpc_peering_connection_id = "${aws_vpc_peering_connection.peer.id}"
- *   auto_accept               = true
- * 
- *   tags = {
- *     Side = "Accepter"
- *   }
- * }
- * 
- * resource "aws_vpc_peering_connection_options" "requester" {
- *   provider = "aws.requester"
- * 
- *   # As options can't be set until the connection has been accepted
- *   # create an explicit dependency on the accepter.
- *   vpc_peering_connection_id = "${aws_vpc_peering_connection_accepter.peer.id}"
- * 
- *   requester {
- *     allow_remote_vpc_dns_resolution = true
- *   }
- * }
- * 
- * resource "aws_vpc_peering_connection_options" "accepter" {
- *   provider = "aws.accepter"
- * 
- *   vpc_peering_connection_id = "${aws_vpc_peering_connection_accepter.peer.id}"
- * 
- *   accepter {
- *     allow_remote_vpc_dns_resolution = true
- *   }
- * }
- * ```
  */
 export class PeeringConnectionOptions extends pulumi.CustomResource {
     /**
```

Not sure why we're dropping the second example entirely, but the first is getting picked up and converted.

Comment (jen20) [reply sdk/go/aws/ec2/peeringConnectionOptions.go]: This is locking the latest version of `pulumi-terraform`: https://github.com/pulumi/pulumi-aws/pull/462/files#diff-bd247e83efc3c45ae9e8c47233249f18R840 - is there a particular change you expect to change this?

Comment (pgavlin) [reply sdk/go/aws/ec2/peeringConnectionOptions.go]: Hm, interesting. I would have expected this to get picked up, then.

Let me cut a new build of `tf2pulumi` and we can retry.

Comment (jen20) [reply sdk/go/aws/ec2/peeringConnectionOptions.go]: Updating both tf2pulumi and pulumi-terraform to the (new) latest have certainly helped here.

**Pull request merged**